api/nodes/serializers.py

```diff
@@ -266,6 +269,111 @@ def update(self, node, validated_data):
         return node
 
 
+class NodeAddonSettingsSerializer(JSONAPISerializer):
+    """
+    Overrides AddonSettingsSerializer to return node-specific fields
```

Comment:
This says it overrides AddonSettingsSerializer but it subclasses from JSONAPISerializer.